packages/kbd/src/Kbd/Kbd.hooks.tsx

```diff
@@ -0,0 +1,120 @@
+import React from "react";
+
+type PressedContextState = {
+  currentlyPressedKeys: Array<string> | null;
+};
+
+const PressedContext = React.createContext<PressedContextState>({
+  currentlyPressedKeys: null,
+});
+
+type UsePressedParams = {
+  key: KeyboardEvent["key"];
+};
+
+const usePressed = ({ key }: UsePressedParams) => {
+  const context = React.useContext(PressedContext);
+  const [pressed, setPressed] = React.useState(false);
+
+  const handleKeyDown = React.useCallback(
+    (event: KeyboardEvent) => {
+      if (event.key === key) {
+        setPressed(true);
+      }
+    },
+    [key],
+  );
+
+  const handleKeyUp = React.useCallback(
+    (event: KeyboardEvent) => {
+      if (event.key === key) {
+        setPressed(false);
+      }
+    },
+    [key],
+  );
+
+  React.useEffect(() => {
```

Comment:
just curious, why do you use `React.useEffect` (and useState etc) instead of importing `useEffect` and using it without the preface?

Reply:
For a lot of these I end up importing a lot of types from react so it's just cleaner to import top level `React` here. Also a personal preference as I find most instances of destructing leading to hard to read/less verbose code.